### ISSUE TITLE
Different lxc-ls output lead to 500

### ISSUE THREAD

**Issue by shadycuz** [claudyus/LXC-Web-Panel#106]
Yeah ran some updates, Ubuntu 14.04, I had been up to date. Saw 7 packages and ran them, not sure   what the packages were maybe lxc? Not sure how to trouble shoot it. Web panel was running fine until them. Much better then the Official panel I might add =)

Oh I forgot to say, that when I visit the panel I get the login box, but when I login I get the error =( and I restarted the server, and stopped started the service. Not sure what else to try.

**Comment by claudyus** [comment]
Try the --debug option as per Readme and paste the exception here.

**Comment by shadycuz** [comment]
Sorry if I'm a bit dense, but I installed using the sudo apt-get install lwp using your ppa or something similar, Do I need to make a clone of the repository in order to run the ./bin/lwp -debug ?

**Comment by shadycuz** [comment]
also I went into config and set debug = true

**Comment by shadycuz** [comment]
I'm so noob, tried to login again...

KeyError
KeyError: 'Webmin'

Traceback (most recent call last)
File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1701, in __call__
return self.wsgi_app(environ, start_response)
File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1689, in wsgi_app
response = self.make_response(self.handle_exception(e))
File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1687, in wsgi_app
response = self.full_dispatch_request()
File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1360, in full_dispatch_request
rv = self.handle_user_exception(e)
File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1358, in full_dispatch_request
rv = self.dispatch_request()
File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1344, in dispatch_request
return self.view_functions[rule.endpoint](**req.view_args)
File "/usr/lib/python2.7/dist-packages/lwp/utils.py", line 94, in new_handler
return handler(*args, **kwargs)
File "/usr/lib/python2.7/dist-packages/lwp/views/main.py", line 40, in home
listx = lxc.listx()
File "/usr/lib/python2.7/dist-packages/lwp/lxclite/__init__.py", line 127, in listx
if status_container[container][0] == 'RUNNING':
KeyError: 'Webmin'
The debugger caught an exception in your WSGI application. You can now look at the traceback which led to the error.
To switch between the interactive traceback and the plaintext one, you can click on the "Traceback" headline. From the text traceback you can also create a paste of it. For code execution mouse-over the frame you want to debug and click on the console icon on the right side.

You can execute arbitrary Python code in the stack frames and there are some extra helpers available for introspection:

dump() shows all variables in the frame
dump(obj) dumps all that's known about the object

**Comment by shadycuz** [comment]
https://gist.github.com/anonymous/e6d68194e06942d5e6e6

**Comment by claudyus** [comment]
I should try to replicate the problem. The traceback tell use that your Webmin container has something different beside the other one.
Can you check if the "Webmin" directory (inside /var/lib/lxc) exists? The directory is a valid container (it has a config file inside)?
Can you also paste on a gist che output of ``` sudo  lxc-ls --fancy``` and ``` ls /var/lib/lxc```

**Comment by shadycuz** [comment]
shadycuz@Web:~$ sudo lxc-ls -f
NAME   STATE   AUTOSTART GROUPS IPV4                                     IPV6
Webmin RUNNING 1         -      10.0.4.186, 10.0.4.3, 10.0.4.4, 10.0.4.5 -
nginx  RUNNING 1         -      10.0.4.2, 10.0.4.200    


shadycuz@Web:~$ sudo ls /var/lib/lxc
nginx  Webmin

not sure where the 10.0.4.200 and 10.0.3.186 came from, the rest I set with static.

**Comment by shadycuz** [comment]
should I uninstall and reinstall or are you working on a fix? Maybe you need beer and pizza money? I miss my Webpanel =(

**Comment by jf1603** [comment]
Is it probably the same like this here: https://github.com/lxc-webpanel/LXC-Web-Panel/issues/36

**Comment by shadycuz** [comment]
From what I read on that, it had to do with log files being in /var/lib/lxc but I don't have any weird files, just my 2 container folders.

**Comment by shadycuz** [comment]
I ran through the install again. Using https://packagecloud.io/claudyus/LXC-Web-Panel/packages/ubuntu/trusty/lwp_0.9_all.deb

I'm pretty sure it was fresh install because it made me use the default username name and password

KeyError
KeyError: 'Webmin'

```
Traceback (most recent call last)
File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1701, in __call__
return self.wsgi_app(environ, start_response)
File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1689, in wsgi_app
response = self.make_response(self.handle_exception(e))
File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1687, in wsgi_app
response = self.full_dispatch_request()
File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1360, in full_dispatch_request
rv = self.handle_user_exception(e)
File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1358, in full_dispatch_request
rv = self.dispatch_request()
File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1344, in dispatch_request
return self.view_functions[rule.endpoint](**req.view_args)
File "/usr/lib/python2.7/dist-packages/lwp/utils.py", line 95, in new_handler
return handler(*args, **kwargs)
File "/usr/lib/python2.7/dist-packages/lwp/views/main.py", line 41, in home
listx = lxc.listx()
File "/usr/lib/python2.7/dist-packages/lwp/lxclite/__init__.py", line 127, in listx
if status_container[container][0] == 'RUNNING':
KeyError: 'Webmin'
The debugger caught an exception in your WSGI application. You can now look at the traceback which led to the error.
To switch between the interactive traceback and the plaintext one, you can click on the "Traceback" headline. From the text traceback you can also create a paste of it. For code execution mouse-over the frame you want to debug and click on the console icon on the right side.

You can execute arbitrary Python code in the stack frames and there are some extra helpers available for introspection:

dump() shows all variables in the frame
dump(obj) dumps all that's known about the object


Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1701, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1689, in wsgi_app
    response = self.make_response(self.handle_exception(e))
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1687, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1360, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1358, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1344, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/lib/python2.7/dist-packages/lwp/utils.py", line 95, in new_handler
    return handler(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/lwp/views/main.py", line 41, in home
    listx = lxc.listx()
  File "/usr/lib/python2.7/dist-packages/lwp/lxclite/__init__.py", line 127, in listx
    if status_container[container][0] == 'RUNNING':
KeyError: 'Webmin'
```

**Comment by shadycuz** [comment]
Gist https://gist.github.com/anonymous/90baf456c59e71d970b6

**Comment by claudyus** [comment]
@shadycuz if you want to workaround the problem you can surrounding the if statements at line 127 of ```/usr/lib/python2.7/dist-packages/lwp/lxclite/__init__.py``` with a try/except, something like:
```
for container in ls():
  try:
     if .....
       ...
  except:
     pass
```

Anyway I cannot replicate the bug and until we don't undestand why "Webmin" container lead to this bug we cannot propose a patch. The PR that @jf1603 suggest has a similar fix also in this fork, https://github.com/claudyus/LXC-Web-Panel/blame/master/lwp/lxclite/__init__.py#L104 and I don't see much differences here.

**Comment by swoopsta** [comment]
I'm not a versed in python at all but by changing the value in '-n+X' at line
`outcmd = _run('lxc-ls --fancy | tail -n+3', output=True).splitlines()`
I've got a machine running 5 containers and it will list a single container in the overview when I change that value. It will, however, list all containers in the left sidebar. Not sure if this helps debug the issue or not, but I thought I'd add what I'm seeing as I'm also coming across this bug with Ubuntu Server 15.10.

**Comment by claudyus** [comment]
@swoopsta comment point this discussion in the correct direction.
@shadycuz, @swoopsta  when I run the lxc-ls command I get this output.
```
 sudo lxc-ls --fancy
NAME  STATE    IPV4  IPV6  GROUPS  AUTOSTART  
--------------------------------------------
base  STOPPED  -     -     -       NO         
test  STOPPED  -     -     -       NO     
```

I just notice that in @shadycuz's comment few days ago the 2nd dashed line is missed! I'm using lxc 1.1.2 under ubuntu. Probabily you are using another version/os anyway this output diff could be managed at lxclite level with a patch. I will propose a patch soon.

**Comment by alroger** [comment]
Thanks, the workaround solved it for me.

**Comment by Artanicus** [comment]
The same workaround fixes this for me as well with  lxc 2.0.1 running on Ubuntu 16.04. Should be pushed into a release methinks.
